packages/arb-token-bridge-ui/tests/e2e/specs/withdrawERC20.cy.ts

```diff
@@ -9,244 +9,259 @@ import {
-      cy.findMoveFundsButton().should('be.disabled')
-      cy.findSelectTokenButton('ETH')
-    })
-
-    it('should withdraw ERC-20 to the same address successfully', () => {
-      ERC20AmountToSend = Number((Math.random() * 0.001).toFixed(5)) // randomize the amount to be sure that previous transactions are not checked in e2e
-
-      cy.login({ networkType: 'childChain' })
-      context('should add ERC-20 correctly', () => {
-        cy.searchAndSelectToken({
-          tokenName: 'WETH',
-          tokenAddress: l2WethAddress
-        })
-      })
+  Object.keys(withdrawalTestCases).forEach(tokenType => {
+    const testCase = withdrawalTestCases[tokenType]
+    // Happy Path
+    context(`User is on L2 and imports ${tokenType}`, () => {
+      let l1ERC20bal: string, l2ERC20bal: string
+
+      // log in to metamask before withdrawal
+      beforeEach(() => {
+        getInitialERC20Balance({
+          tokenAddress: testCase.l1Address,
+          multiCallerAddress: getL1NetworkConfig().multiCall,
+          address: Cypress.env('ADDRESS'),
+          rpcURL: Cypress.env('ETH_RPC_URL')
+        }).then(
+          val =>
+            (l1ERC20bal = formatAmount(val, {
+              symbol: testCase.symbol
+            }))
+        )
 
-      context('should show summary', () => {
-        cy.typeAmount(ERC20AmountToSend)
-        cy.findGasFeeSummary(zeroToLessThanOneETH)
-        cy.findGasFeeForChain(getL2NetworkName(), zeroToLessThanOneETH)
-        cy.findGasFeeForChain(
-          new RegExp(
-            `You'll have to pay ${getL1NetworkName()} gas fee upon claiming.`,
-            'i'
-          )
+        getInitialERC20Balance({
+          tokenAddress: testCase.l2Address,
+          multiCallerAddress: getL2NetworkConfig().multiCall,
+          address: Cypress.env('ADDRESS'),
+          rpcURL: Cypress.env('ARB_RPC_URL')
+        }).then(
+          val =>
+            (l2ERC20bal = formatAmount(val, {
+              symbol: testCase.symbol
+            }))
         )
       })
 
-      context('should show clickable withdraw button', () => {
-        cy.findMoveFundsButton().click()
+      it('should show form fields correctly', () => {
+        cy.login({ networkType: 'childChain' })
+        cy.findSourceChainButton(getL2NetworkName())
+        cy.findDestinationChainButton(getL1NetworkName())
+        cy.findMoveFundsButton().should('be.disabled')
+        cy.findSelectTokenButton('ETH')
       })
 
-      context('should withdraw successfully', () => {
-        cy.findByText(/Arbitrum’s bridge/i).should('be.visible')
-
-        // the Continue withdrawal button should be disabled at first
-        cy.findByRole('button', {
-          name: /Continue/i
-        }).should('be.disabled')
+      it(`should withdraw ${tokenType} to the same address successfully`, () => {
+        ERC20AmountToSend = Number((Math.random() * 0.001).toFixed(5)) // randomize the amount to be sure that previous transactions are not checked in e2e
```

Comment:
We're running node for each tests so this should not happen

Reply:
Didn't touch this code in current PR, but to justify this - it's present coz 
- we don't want to repeat the ERC20 amount even in the current test trials (3 trials per test). We don't want the new attempt to check for previous value so this is valid.
- withdrawErc20 has different variations of tests - normal / custom destination - and we'd want to check for a new value in every test to avoid `findAllByText`

If still an issue, we can clean this up separately.